public/app/features/explore/spec/queryHistory.test.tsx

```diff
@@ -42,6 +44,60 @@ describe('Explore: Query History', () => {
 
     // previously added query is in query history
     await openQueryHistory();
-    assertQueryHistoryExists(RAW_QUERY);
+    await assertQueryHistoryExists(RAW_QUERY);
+  });
+
+  it('adds recently added query if the query history panel is already open', async () => {
+    const urlParams = {
+      left: serializeStateToUrlParam({
+        datasource: 'loki',
+        queries: [{ refId: 'A', expr: 'query #1' }],
+        range: { from: 'now-1h', to: 'now' },
+      }),
+    };
+
+    const { datasources } = setupExplore({ urlParams });
+    (datasources.loki.query as jest.Mock).mockReturnValueOnce(makeLogsQueryResponse());
+    await waitForExplore();
+    await openQueryHistory();
+
+    inputQuery('query #2');
+    runQuery();
+    await assertQueryHistory(['{"expr":"query #2"}', '{"expr":"query #1"}']);
+  });
+
+  it('updates the state in both Explore panes', async () => {
+    const urlParams = {
+      left: serializeStateToUrlParam({
+        datasource: 'loki',
+        queries: [{ refId: 'A', expr: 'query #1' }],
+        range: { from: 'now-1h', to: 'now' },
+      }),
+      right: serializeStateToUrlParam({
+        datasource: 'loki',
+        queries: [{ refId: 'A', expr: 'query #2' }],
+        range: { from: 'now-1h', to: 'now' },
+      }),
+    };
+
+    const { datasources } = setupExplore({ urlParams });
+    (datasources.loki.query as jest.Mock).mockReturnValue(makeLogsQueryResponse());
+    await waitForExplore();
+    await waitForExplore(ExploreId.right);
+
+    // queries in history
+    await openQueryHistory(ExploreId.left);
+    await assertQueryHistory(['{"expr":"query #2"}', '{"expr":"query #1"}'], ExploreId.left);
+    await openQueryHistory(ExploreId.right);
+    await assertQueryHistory(['{"expr":"query #2"}', '{"expr":"query #1"}'], ExploreId.left);
```

Comment:
is the last argument here correct? are we testing that when opening the right pane, the left pane still has the correct content or should this be 
```suggestion
    await assertQueryHistory(['{"expr":"query #2"}', '{"expr":"query #1"}'], ExploreId.right);
```
?

Reply:
Ooooh good catch. Fixed in a620c07c199acc6851e94d8be093713b026bc3d0